internal/searchindex/searchindex_test.go

```diff
@@ -0,0 +1,514 @@
+						CharFilters:  jsonMustEncode([]map[string]interface{}{{"char": "filter"}, {"char1": "filter1"}}),
+						Tokenizer: akov2.Tokenizer{
+							Type:           pointer.MakePtr("type"),
+							MaxGram:        nil,
+							MinGram:        nil,
+							Group:          nil,
+							Pattern:        nil,
+							MaxTokenLength: nil,
+						},
+					},
+				}),
+				SearchAnalyzer: pointer.MakePtr("searchAnalyzer"),
+				StoredSource:   nil,
+			},
+			ID:     nil,
+			Status: nil,
+		}
+		idx2 := &SearchIndex{
+			SearchIndex: akov2.SearchIndex{
+				Name:           "",
+				DBName:         "",
+				CollectionName: "",
+				Type:           "",
+				Search: &akov2.Search{
+					Synonyms: &([]akov2.Synonym{}),
+					Mappings: &akov2.Mappings{
+						Dynamic: nil,
+						Fields:  nil,
+					},
+					SearchConfigurationRef: common.ResourceRefNamespaced{},
+				},
+				VectorSearch: nil,
+			},
+			AtlasSearchIndexConfigSpec: akov2.AtlasSearchIndexConfigSpec{
+				Analyzer: pointer.MakePtr("analyzer"),
+				Analyzers: &([]akov2.AtlasSearchIndexAnalyzer{
+					{
+						Name:         "test",
+						TokenFilters: jsonMustEncode([]map[string]interface{}{{"token": "filter"}, {"token1": "filter1"}}),
+						CharFilters:  jsonMustEncode([]map[string]interface{}{{"char": "filter"}, {"char1": "filter1"}}),
+						Tokenizer: akov2.Tokenizer{
+							Type:           pointer.MakePtr("type"),
+							MaxGram:        nil,
+							MinGram:        nil,
+							Group:          nil,
+							Pattern:        nil,
+							MaxTokenLength: nil,
+						},
+					},
+				}),
+				SearchAnalyzer: pointer.MakePtr("searchAnalyzer"),
+				StoredSource:   nil,
+			},
+			ID:     nil,
+			Status: nil,
+		}
+		isEqual, err := idx1.EqualTo(idx2)
+		assert.Nil(t, err)
+		assert.True(t, isEqual)
+	})
+}
+
+//nolint:dupl
+func TestSearchIndex_ToAtlas(t *testing.T) {
+	type fields struct {
+		SearchIndex                akov2.SearchIndex
+		AtlasSearchIndexConfigSpec akov2.AtlasSearchIndexConfigSpec
+		ID                         *string
+		Status                     *string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    *admin.ClusterSearchIndex
+		wantErr bool
+	}{
+		{
+			name: "Should convert to index to Atlas using a valid index",
+			fields: fields{
+				SearchIndex: akov2.SearchIndex{
+					Name:           "name",
+					DBName:         "db",
+					CollectionName: "collection",
+					Type:           "search",
+					Search: &akov2.Search{
+						Synonyms: &([]akov2.Synonym{
+							{
+								Name:     "name",
+								Analyzer: "analyzer",
+								Source:   akov2.Source{Collection: "collection"},
+							},
+						}),
+						Mappings: &akov2.Mappings{
+							Dynamic: pointer.MakePtr(true),
+							Fields:  jsonMustEncodeMap(map[string]interface{}{"field": "value"}),
+						},
+						SearchConfigurationRef: common.ResourceRefNamespaced{},
+					},
+					VectorSearch: &akov2.VectorSearch{},
```

Comment:
I believe the e2e test failing is missing some configuration here:

`https://cloud-qa.mongodb.com/api/atlas/v2/groups/6634bc7dee5df86c35d9f7c2/clusters/search-nodes-test/fts/indexes POST: HTTP 400 Bad Request (Error code: "MISSING_ATTRIBUTE") Detail: The required attribute [fields] was not specified. Reason: Bad Request. Params: [[fields]]`

Will try to figure out a simple fields config that would work.

Reply:
nope not here...